source/getting-started/control-with-amdc/encoder-fb/index.md

```diff
@@ -2,85 +2,96 @@
+
+Example code to convert encoder to angular position in radians:
 ```C
 double task_get_theta_m(void)
 {
     // Get raw encoder position
     uint32_t position;
     encoder_get_position(&position);
 
-    // Encoder count per revolution
+    int ENCODER_COUNT_PER_REV, CCW;
+    double enc_theta_m_offset;
+
+    // User to set encoder count per revolution
     ENCODER_COUNT_PER_REV = 1024;
 
+    // Set 1 if encoder count increases with CCW rotation of shaft, Set 0 if encoder count increases with CW rotation of shaft
+    CCW = 1; 
+
     // Angular position to be computed
     double theta_m_enc;
 
+    // User to set encoder offset
     enc_theta_m_offset = 100;
 
-    // Convert to radians
-    theta_m_enc = (double) PI2 * ( ( (double)position - enc_theta_m_offset )/ (double) ENCODER_COUNT_PER_REV);
 
-    // Wrapping to ensure within bounds
-    while (theta_m_enc < 0) {
-        theta_m_enc += PI2;
+    // Convert to radians
+    if (CCW){
+        theta_m_enc = (double) PI2 * ( ( (double)position - enc_theta_m_offset )/ (double) ENCODER_COUNT_PER_REV);
+    }
+    else{
+        theta_m_enc = (double) PI2 * ( ( (double)ENCODER_COUNT_PER_REV - (double)1 -(double)position + enc_theta_m_offset )/ (double) ENCODER_COUNT_PER_REV);
     }
+    
+     // Mod by 2 pi
+    theta_m_enc = fmod(theta_m_enc,PI2 );
     return theta_m_enc;
 }
 ```
 
-Example code for when encoder count is decreasing with CCW rotation of shaft:
-```C
-double task_get_theta_m(void)
-{
-    // Get raw encoder position
-    uint32_t position;
-    encoder_get_position(&position);
 
-    // Encoder count per revolution
-    ENCODER_COUNT_PER_REV = 1024;
 
-    // Angular position to be computed
-    double theta_m_enc;
+### Finding the offset
 
-    enc_theta_m_offset = 100;
+The example code shown above makes uses of an encoder offset value. It is necessary for the user to find this offset experimentally for their machine. For synchronous machines, this offset is the count value measured by the encoder when the d-axis of the rotor is aligned with the phase U winding of the stator. 
 
-    // Convert to radians
-    theta_m_enc = (double) PI2 * ( ( (double)ENCODER_COUNT_PER_REV - (double)1 -(double)position + enc_theta_m_offset )/ (double) ENCODER_COUNT_PER_REV);
+To get the rotor to align with the phase U winding, the user may have some current flow through phase U and out of phase V and phase W. This can done by using a DC supply, with phase U connected to the positive terminal and phases V and W connected to the negative terminal. Alternately, the user may also inject some current along the d-axis using the AMDC injection function, but with the rotor position overridden by the user to 0. After obtaining the offset, the user needs to set the variable `enc_theta_m_offset` to the appropirate value in the `task_get_theta_m()` function. 
```

Comment:
Add more commentary about "AMDC injection function"....a user must have already set this up in your control code...

Reply:
How does someone "override the rotor position to 0" ?

Reply:
"appropirate" typo